omod/src/main/java/org/openmrs/module/adminui/page/controller/metadata/privileges/ManagePrivilegesPageController.java

```diff
@@ -22,5 +38,33 @@
 	public void get(PageModel model, @SpringBean("userService") UserService userService) {
 		model.addAttribute("privileges", userService.getAllPrivileges());
 	}
-	
+
+
+    public String post(PageModel model, 
+            		   @RequestParam(value = "privilegeName", required = false) @BindParams Privilege privilege,
+                       @RequestParam("action") String action,
+                       @RequestParam(value = "reason", required = false) String reason,
+                       @SpringBean("userService") UserService userService, HttpServletRequest request) {
+
+    	Errors errors = new BeanPropertyBindingResult(privilege, "privilege");
+        new PrivilegeValidator().validate(privilege, errors);
```

Comment:
There is no point in validating an item you are going to purge from the database